### ISSUE TITLE
`msiDataObjCopy` not triggering any pep

### ISSUE THREAD

**Issue by mstfdkmn** [irods/irods#7387]
### iRODS Version, OS and Version
4.3.0, almalinux8

### What did you try to do?
I tried to copy a data object through ` msiDataObjCopy` both in the NREP and in the PREP.
 
### Expected behavior
I wanted to see `pep_api_data_obj_copy_post` or any other pep is triggered when ` msiDataObjCopy` copies an object.

### Observed behavior (including steps to reproduce, if applicable)
Data objects are copied by ` msiDataObjCopy` but no relevant PEP is firing.

PREP:
```
[irods@dev-u0137480 irods]$ cat test.r
def main(rule_args, callback, rei):

    callback.msiDataObjCopy("/u0137480/home/rods/test.txt", "/u0137480/home/rods/coll_A/test.txt", \
                                        "forceFlag=", 0)
INPUT null
OUTPUT ruleExecOut
```

NREP:
```
[irods@dev-u0137480 irods]$ cat rule_copy_msi.r
myTestRule {
  msiDataObjCopy(*SourceFile,*DestFile,"forceFlag=",*Status);
  writeLine("stdout","File *SourceFile copied to *DestFile");
}
INPUT *SourceFile="/u0137480/home/rods/foo1", *DestFile="/u0137480/home/rods/sub2/foo1"
OUTPUT ruleExecOut
```

**Comment by trel** [comment]
Microservices (and Python functions in PREP) run *in the server* so will not trigger any API PEPs.

**Comment by mstfdkmn** [comment]
As a relevant pep, in the documentation I can only see `pep_api_data_copy_post` and `pep_resource_copy_post` other than API pep. But they are not triggered either by `msiDataObjCopy`. Also, fyi this `msiDataObjPut()` triggers its counterpart `pep_api_data_obj_put_post`.

But okay we will probably block usage of those msi(s) that dont fire any pep. Thanks.

**Comment by korydraughn** [comment]
On the `msiDataObjPut`, is a redirect happening?

That is, is the data (at rest) on the same computer where `msiDataObjPut` was invoked?

**Comment by mstfdkmn** [comment]
Yes. I tested in a container environment where irods server and data exist.

**Comment by korydraughn** [comment]
How many iRODS servers are in the container environment you tested with?

**Comment by mstfdkmn** [comment]
We have one irods server.

**Comment by korydraughn** [comment]
I've confirmed that `msiDataObjCopy` does not trigger API PEPs while in the server. That's expected.

`msiDataObjPut` seems to be special. The following comes from its [4.3.3 documentation](https://docs.irods.org/4.3.3/doxygen/reDataObjOpr_8cpp.html#a8076987f48ddb90fdfc0a7f5c5dcf13b):

> Note
>
> This call should only be used through the rcExecMyRule (irule) call i.e., rule execution initiated by clients and should not be called internally by the server since it interacts with the client through the normal client/server socket connection. Also, it should never be called through delayExec since it requires client interaction.

`msiDataObjPut` intentionally does not follow the normal pattern and leaves me wondering why it's designed with such requirements. Seems easy to misuse. Also, what does it solve?

@trel Thoughts?

Seems to me `msiDataObjPut` deserves its own issue, allowing us to close this one since `msiDataObjCopy` works in the expected way.

**Comment by trel** [comment]
Agreed on closing this one and opening a new issue.

I expect `msiDataObjPut` is different because over the years it was desired that, along with `acPostProcForPut`, the 'new file landing in the system' use case was required/asked to handle all manner of different scenarios ... and became inconsistent.  This is one of the reasons we moved towards dynamic PEPs overall, to be better able to assert and understand when things were firing.

**Comment by korydraughn** [comment]
That makes sense.

I've opened #8229 to address `msiDataObjPut`.
